qiskit/transpiler/target.py

```diff
@@ -1311,151 +1314,218 @@ def from_configuration(
+                                    "falling back to `BackendProperties` to fetch duration",
+                                    RuntimeWarning,
+                                )
+                                duration = (
+                                    params["gate_length"][0] if "gate_length" in params else None
+                                )
+                        else:
+                            duration = params["gate_length"][0] if "gate_length" in params else None
+
+                        if prop_name_map[name] is None:
+                            prop_name_map[name] = {}
+
+                        prop_name_map[name][qubits] = InstructionProperties(
+                            error=error, duration=duration
+                        )
 
-                    if error is None and duration is None and calibration is None:
-                        gate_properties[edge] = None
-                    else:
-                        gate_properties[edge] = InstructionProperties(
-                            duration=duration, error=error, calibration=calibration
+                        if isinstance(prop_name_map[name], dict) and any(
+                            v is None for v in prop_name_map[name].values()
+                        ):
+                            # Properties provides gate properties only for subset of qubits
+                            # Associated qubit set might be defined by the gate config here
+                            logger.info(
+                                "Gate properties of instruction %s are not provided for every qubits. "
+                                "This gate is ideal for some qubits and the rest is with finite error. "
+                                "Created backend target may confuse error-aware circuit optimization.",
+                                name,
+                            )
+                except BackendPropertyError:
+                    # This gate doesn't report any property
+                    continue
+
+            # Measure instruction property is stored in qubit property
+            prop_name_map["measure"] = {}
+            for qubit_idx in range(num_qubits):
+                qubit_prop = backend_properties.qubit_property(qubit_idx)
+                duration = (
+                    qubit_prop["readout_length"][0] if "readout_length" in qubit_prop else None
+                )
+                error = qubit_prop["readout_error"][0] if "readout_error" in qubit_prop else None
+                prop_name_map["measure"][(qubit_idx,)] = InstructionProperties(
+                    duration=duration, error=error
+                )
+
+        if num_qubits is not None:
+            for op in required_operations:
+                # Map required ops to each operational qubit
+                if prop_name_map[op] is None:
+                    prop_name_map[op] = {
+                        (q,): None for q in range(num_qubits) if q not in faulty_qubits
```

Comment:
This is bit fragile because there is no guarantee that `op` is always single qubit operation.

Reply:
removed the logic all together,
[humble suggestion]:
when we will add something which is not a single qubit operation in `required_operations`, then in the same PR we will deal with this at that time.